src/query-builder/transformer/RawSqlResultsToEntityTransformer.ts

```diff
@@ -393,61 +390,84 @@ export class RawSqlResultsToEntityTransformer {
+                        .databaseName // todo: fix joinColumns[0]
+            } else {
+                referenceColumnName = relation.isOwning
+                    ? relation.joinColumns[0].referencedColumn!.databaseName
+                    : relation.inverseRelation!.joinColumns[0].referencedColumn!
+                          .databaseName
+            }
 
-                const referenceColumnValue =
-                    rawSqlResults[0][
-                        DriverUtils.buildAlias(
-                            this.driver,
-                            undefined,
-                            alias.name,
-                            referenceColumnName,
-                        )
-                    ] // we use zero index since its grouped data // todo: selection with alias for entity columns wont work
-                if (
-                    referenceColumnValue !== undefined &&
-                    referenceColumnValue !== null
-                ) {
+            const referenceColumnValue =
+                rawSqlResults[0][
+                    this.buildAlias(alias.name, referenceColumnName)
+                ] // we use zero index since its grouped data // todo: selection with alias for entity columns wont work
+            if (
+                referenceColumnValue !== undefined &&
+                referenceColumnValue !== null
+            ) {
+                entity[
+                    rawRelationCountResult.relationCountAttribute.mapToPropertyPropertyName
+                ] = 0
+                for (const result of rawRelationCountResult.results) {
+                    if (result["parentId"] !== referenceColumnValue) continue
                     entity[
                         rawRelationCountResult.relationCountAttribute.mapToPropertyPropertyName
-                    ] = 0
-                    rawRelationCountResult.results
-                        .filter(
-                            (result) =>
-                                result["parentId"] === referenceColumnValue,
-                        )
-                        .forEach((result) => {
-                            entity[
-                                rawRelationCountResult.relationCountAttribute.mapToPropertyPropertyName
-                            ] = parseInt(result["cnt"])
-                            hasData = true
-                        })
+                    ] = parseInt(result["cnt"])
+                    hasData = true
                 }
-            })
+            }
+        }
 
         return hasData
     }
 
+    private getColumnsToProcess(aliasName: string, metadata: EntityMetadata) {
+        let metadatas = this.columnsCache.get(aliasName)
+        if (!metadatas) {
+            metadatas = new Map()
+            this.columnsCache.set(aliasName, metadatas)
+        }
+        let columns = metadatas.get(metadata)
+        if (!columns) {
+            columns = metadata.columns
+                .filter(
```

Comment:
In the other sections it was more about removing `forEach` for clearer profiling (dropping `filter` in a couple of instances was a side effect). I'm not sure it's worth the loss in readability in this case since it's well-bounded (there's a small, constant number of columns) and this code is only run once before being cached.